caom2-pkg-server/src/main/java/ca/nrc/cadc/caom2/pkg/Caom2PackageRunner.java

```diff
@@ -0,0 +1,272 @@
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*  $Revision: 5 $
+*
+************************************************************************
+*/
+
+package ca.nrc.cadc.caom2.pkg;
+
+import org.opencadc.pkg.server.PackageItem;
+import org.opencadc.pkg.server.PackageRunner;
+import org.opencadc.pkg.server.PackageRunnerException;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.caom2.Artifact;
+import ca.nrc.cadc.caom2.ObservationURI;
+import ca.nrc.cadc.caom2.PlaneURI;
+import ca.nrc.cadc.caom2.ProductType;
+import ca.nrc.cadc.caom2.PublisherID;
+import ca.nrc.cadc.caom2.artifact.resolvers.CaomArtifactResolver;
+import ca.nrc.cadc.caom2ops.ArtifactQueryResult;
+import ca.nrc.cadc.caom2ops.CaomTapQuery;
+import ca.nrc.cadc.caom2ops.ServiceConfig;
+import ca.nrc.cadc.log.WebServiceLogInfo;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.util.StringUtil;
+import ca.nrc.cadc.uws.ErrorSummary;
+import ca.nrc.cadc.uws.ErrorType;
+import ca.nrc.cadc.uws.ParameterUtil;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URISyntaxException;
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+import javax.security.auth.Subject;
+import org.apache.log4j.Logger;
+
+
+public class Caom2PackageRunner extends PackageRunner {
+    private static final Logger log = Logger.getLogger(Caom2PackageRunner.class);
+
+    private final URI tapID;
+    private List<String> idList;
+    private String packageName;
+
+    public Caom2PackageRunner()
+    { 
+        ServiceConfig sc = new ServiceConfig();
+        this.tapID = sc.getTapServiceID();
+    }
+
+    @Override
+    protected String getPackageName() {
+        return this.packageName;
+    }
+
+
+    @Override
+     public Iterator<PackageItem> getItems() throws IOException, PackageRunnerException {
+
+        // obtain credentials from CDP if the user is authorized
+        AccessControlContext accessControlContext = AccessController.getContext();
+        Subject subject = Subject.getSubject(accessControlContext);
+        AuthMethod authMethod = AuthenticationUtil.getAuthMethod(subject);
+
+        String runID = this.job.getID();
+        if (this.job.getRunID() != null) {
+            runID = this.job.getRunID();
+        }
+
+        List<String> idList = ParameterUtil.findParameterValues("ID", this.job.getParameterList());
+
+        CaomTapQuery query = new CaomTapQuery(tapID, runID);
+
+        CaomArtifactResolver artifactResolver = new CaomArtifactResolver();
+        artifactResolver.setAuthMethod(authMethod); // override auth method for proxied calls
```

Comment:
I don't see any call to `CredUtil.checkCredentials()`, but that has to happen somewhere (here: because CaomTapQuery does remote calls). However, since the PackageRunner base class actually does the HttpGet calls to get the file content, it should probably have the call to `CredUtil.checkCredentials()`. It wouldn't be wrong to also include that call here since this code also does remote calls.

Second, this is not the right AuthMethod to set here. It has to be the one that matches credentials in the subject rather than the one the client used (eg. `AuthenticationUtil.getAuthMethodFromCredentials(subject)`).

Reply:
This is almost funny - in the old code it was called 'proxyAuthMethod' and in a previous review I was asked to remove the 'extra' getAuthMethodFromCredentials call. Right, putting it back in.

Reply:
added back in